src/controller/CHIPDeviceController.cpp

```diff
@@ -870,6 +870,7 @@ CHIP_ERROR DeviceCommissioner::PairDevice(NodeId remoteDeviceId, RendezvousParam
     mDeviceBeingPaired = GetInactiveDeviceIndex();
     VerifyOrExit(mDeviceBeingPaired < kNumMaxActiveDevices, err = CHIP_ERROR_NO_MEMORY);
     device = &mActiveDevices[mDeviceBeingPaired];
+    device->SetCSRNonce(params.GetCSRNonce());
```

Comment:
If you add `HasCSRNonce()` to `RendezvoudParameters`, you can do the following.

```
if (params.HasCSRNonce())
{
    device->SetCSRNonce(params.GetCSRNonce());
}
else
{
    device->GenerateCSRNonce();
}
```

Reply:
Done